mapfile.c

```diff
@@ -4273,10 +4275,22 @@ int loadLayer(layerObj *layer, mapObj *map)
         break;
       case(MS_PLUGIN): {
         int rv;
-        if(getString(&layer->plugin_library_original) == MS_FAILURE) return(-1);
-        rv = msBuildPluginLibraryPath(&layer->plugin_library,
-                                      layer->plugin_library_original,
-                                      map);
+        if(map->config) { // value *must* represent a config key
+          char *value = NULL;
+          const char *plugin_library = NULL;
+
+	  if(getString(&value) == MS_FAILURE) return(-1);
+          plugin_library = msConfigGetPlugin(map->config, value);
+	  msFree(value);
+          if(!plugin_library) {
+            msSetError(MS_MISCERR, "Plugin value not found in config file. See mapserver.org/config_file.html for more information." , "loadLayer()");
```

Comment:
In https://github.com/MapServer/MapServer-documentation/pull/613 the URL would be mapserver.org/mapfile/config.html (seemed to fit alongside other Mapfile keyword docs - happy to change though)

Reply:
The URL is just a placeholder so I'm not set on it and we can certainly change it. However, the config file is distinct from a mapfile so perhaps they should live at the same level? That is, perhaps /mapfile and /config_file (or /configfile) as sub-directories, so the URL referenced in code could be mapserver.org/config_file/ which would allow for multiple documents to be stored in that section.

Reply:
Although distinct I couldn't think of much else to write, or split into more than one page: https://github.com/MapServer/MapServer-documentation/blob/090a233ba2f81156b74860470e8a441414b833c1/en/mapfile/config.txt
Could maybe be moved to the /installation/ folder?

Reply:
I could see moving a couple of pages into that section, for example:

  https://mapserver.org/optimization/environment_variables.html
  https://mapserver.org/optimization/limit_mapfile_access.html

Those shouldn't be under optimization when they're core concepts.

--Steve